### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by shagedorn**
* Sample app uses the dynamic framework
* Sample app now uses all categories (in the same order the categories are defined), unless they require capabilities – those are added, too, but commented out. This should allow us to easily test all categories locally.
* Add required configuration for Health and TencentWeibo requests in sample app
* Xcode housekeeping

### CODE REVIEW

File: PermissionsTestApp/AppDelegate.m [hunk 1]
```diff
@@ -32,25 +34,45 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
...
              @(ISHPermissionCategoryPhotoCamera),
              @(ISHPermissionCategoryNotificationLocal),
-             @(ISHPermissionCategorySocialTwitter),
+             // requires Push capability & entitlements to actually work
+             @(ISHPermissionCategoryNotificationRemote),
+
              @(ISHPermissionCategorySocialFacebook),
+             @(ISHPermissionCategorySocialTwitter),
+             @(ISHPermissionCategorySocialSinaWeibo),
+             // TODO: alert cannot be presented
```
Comment (shagedorn) [inline PermissionsTestApp/AppDelegate.m]: This is https://github.com/iosphere/ISHPermissionKit/issues/72

File: ISHPermissionKit/Requests/ISHPermissionRequestAccount.m [hunk 2]
```diff
@@ -38,6 +48,7 @@ - (NSString *)accountTypeIdentifier {
             return ACAccountTypeIdentifierTencentWeibo;
             
         default:
+            NSAssert(NO, @"Invalid category: %@", @(self.permissionCategory));
```
Comment (felixLam) [inline ISHPermissionKit/Requests/ISHPermissionRequestAccount.m]: Shouldn't this follow the same pattern as above without a default?

**Comment by felixLam** [comment]
:shipit:

File: ISHPermissionKit/Requests/ISHPermissionRequestAccount.m (hunk 2 above)
Comment (shagedorn) [inline ISHPermissionKit/Requests/ISHPermissionRequestAccount.m]: The code above probably raises a warning... we don't really want to list all ~15 categories, so I believe `default` is ok here. Plus, we usually don't care here if new categories are added.

**Comment by felixLam** [comment]
:shipit:

**Pull request merged**